app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -144,6 +141,71 @@ public void afterTextChanged(Editable s) {
         }
     }
 
+    // Return true is system wide dark theme is enabled else false
+    private boolean getSystemDefaultThemeBool(String theme) {
+        switch (theme) {
+            case "Dark":
+                return true;
+            case "Default":
+                return getSystemDefaultThemeBool(getSystemDefaultTheme());
+            default:
+                return false;
+        }
+    }
+
+    // Returns the default system wide theme
+    private String getSystemDefaultTheme() {
+        if ((getResources().getConfiguration().uiMode &
+                Configuration.UI_MODE_NIGHT_MASK) == Configuration.UI_MODE_NIGHT_YES) {
+            return "Dark";
+        }
+        return "Light";
+    }
+
+    /**
+     * Prepares themes list and adds them to list preference as pairs.
+     * Uses previously saved theme if there is any, if not then uses default.
+     * Adds preference theme listener and saves value choosen by user to shared preferences
+     * to remember later
+     */
+    private void prepareTheme() {
+
+        String[] themeOptionsNames = {"Default", "Dark", "Light"};
```

Comment:
There is no separation between UI and data representation. As such these strings are untranslatable

Reply:
Good point, I missed to catch it, I will move them to strings.xml

Reply:
I would also map the internal representation to enums, switching on strings is a headache most people don't need. 

It would also be preferrable if you could pass this data in in the `preferences.xml`.

Reply:
Please review the new changes and let me know if I understood correctly.